hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+from hs_communities.models import Topic
+
+
+class CollaborateView(TemplateView):
+    template_name = 'pages/collaborate.html'
+
+
+class CommunityView(TemplateView):
+    template_name = 'hs_communities/community.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(CommunityView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        # user_id = User.objects.get(pk=self.request.user.id)
+        grpfilter = self.request.GET.get('grp')
+
+        community_resources = community_from_name_or_id("CZO National Community").public_resources
+        groups = []
+        for c in community_resources:
+            if not any(str(c.group_id) == g.get('id') for g in groups):  # if the group id is not already present in the list
+                if c.group_name != "CZO National":  # The National Group is used to establish the entire Community
+                    res_count = len([r for r in community_resources if r.group_name == c.group_name])
+                    groups.append({'id': str(c.group_id), 'name': str(c.group_name), 'res_count': str(res_count)})
+
+        groups = sorted(groups, key=lambda key: key['name'])
+        return {
+            'community_resources': community_resources,
+            'groups': groups,
+            'grpfilter': grpfilter,
+        }
+
+
+class FindCommunitiesView(TemplateView):
+    template_name = 'hs_communities/find-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(FindCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        return {
+            'communities_list': Community.objects.all()
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
+        """
+        return the community membership information of a group; group can belong to only one community
+        :param grp: group object
+        :param communities: communities
+        :return: tuple id, name of community
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+        # for each group set group dynamic attributes
+        for g in groups:
+            g.is_user_member = u in g.gaccess.members
+            if g.is_user_member:
+                grps_member_of.append(g)
+            g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(request_from=u).exists()
```

Comment:
No reason to set these if the user isn't a member of the gorup as they are discarded.